### ISSUE TITLE
[Flaky Test]: TestStandaloneUpgradeRollback – upgrade details in the state cannot be nil

### ISSUE THREAD

**Issue by rdner** [elastic/elastic-agent#5437]
### Failing test case

TestStandaloneUpgradeRollback

### Error message

upgrade details in the state cannot be nil

### Build

* https://buildkite.com/elastic/elastic-agent-extended-testing/builds/2401
* https://buildkite.com/elastic/elastic-agent-extended-testing/builds/2423
* https://buildkite.com/elastic/elastic-agent-extended-testing/builds/2307

### OS

Linux

### Stacktrace and notes

    upgrader.go:347: waiting for upgrade watcher to start
    upgrader.go:352: upgrade watcher started
    upgrader.go:357: Checking upgrade details state while Upgrade Watcher is running
    fixture.go:657: >> running binary with: [/opt/Elastic/Agent/elastic-agent version --binary-only --yaml]
    fixture.go:657: >> running binary with: [/opt/Elastic/Agent/elastic-agent status --output json]
    fixture.go:657: >> running binary with: [/opt/Elastic/Agent/elastic-agent status --output json]
    upgrader.go:514: waiting for healthy agent and proper version: versions don't match: got 8.16.0, want 8.15.0
    fixture.go:657: >> running binary with: [/opt/Elastic/Agent/elastic-agent status --output json]
    upgrade_rollback_test.go:138: 
        	Error Trace:	/home/ubuntu/agent/testing/integration/upgrade_rollback_test.go:138
        	Error:      	Expected value not to be nil.
        	Test:       	TestStandaloneUpgradeRollback
        	Messages:   	upgrade details in the state cannot be nil
    fixture_install.go:265: [test TestStandaloneUpgradeRollback] Inside fixture cleanup function
    fixture_install.go:281: collecting diagnostics; test failed
    fixture.go:657: >> running binary with: [/opt/Elastic/Agent/elastic-agent diagnostics -f /home/ubuntu/agent/build/diagnostics/TestStandaloneUpgradeRollback-diagnostics-2024-09-04T21:45:16Z.zip]
    fixture.go:657: >> running binary with: [/opt/Elastic/Agent/elastic-agent uninstall --force]
    fixture.go:1028: Dumping running processes in /home/ubuntu/agent/build/diagnostics/TEST-TestStandaloneUpgradeRollback-linux-amd64-ProcessDump-cleanup.json
    fixture.go:1216: Temporary directory "/tmp/TestStandaloneUpgradeRollback653122820" preserved for investigation/debugging
    fixture.go:1216: Temporary directory "/tmp/TestStandaloneUpgradeRollback3477162090" preserved for investigation/debugging
--- FAIL: TestStandaloneUpgradeRollback (111.03s)

**Comment by khushijain21** [comment]
This is still flaky. Another instance 

https://buildkite.com/elastic/elastic-agent/builds/30479#019a96c0-86ef-47eb-b6ff-5bfb44f689d3

Stacktrace
```
 upgrade_rollback_test.go:175:
        	Error Trace:	/opt/buildkite-agent/builds/bk-agent-prod-aws-1763465797685494624/elastic/elastic-agent/testing/integration/ess/upgrade_rollback_test.go:175
        	Error:      	Received unexpected error:
        	            	agent state is not healthy: got 3
        	Test:       	TestStandaloneUpgradeRollback
    fixture_install.go:311: [test TestStandaloneUpgradeRollback] Inside fixture cleanup function
    fixture_install.go:327: collecting diagnostics; test failed
    fixture_install.go:789: >> running binary with: [/opt/Elastic/Agent/elastic-agent diagnostics -f /opt/buildkite-agent/builds/bk-agent-prod-aws-1763465797685494624/elastic/elastic-agent/build/diagnostics/TestStandaloneUpgradeRollback-2025-11-18T12-08-45Z-diagnostics.zip]
    fixture_install.go:740: >> running binary with: [/opt/Elastic/Agent/elastic-agent uninstall --force]
    fixture.go:1160: Dumping running processes in /opt/buildkite-agent/builds/bk-agent-prod-aws-1763465797685494624/elastic/elastic-agent/build/diagnostics/TestStandaloneUpgradeRollback-2025-11-18T12-08-45Z-ProcessDump-cleanup.json
    fixture.go:1451: Temporary directory "/tmp/TestStandaloneUpgradeRollback820357958" preserved for investigation/debugging
    fixture.go:1451: Temporary directory "/tmp/TestStandaloneUpgradeRollback1688710356" preserved for investigation/debugging

```

**Comment by pierrehilbert** [comment]
Hey @khushijain21 this is currently discussed over slack.

**Comment by belimawr** [comment]
Another failure: https://buildkite.com/elastic/elastic-agent/builds/30444/steps/canvas?jid=019a931f-b150-4d4e-b9fe-0c24e22fdb56#019a931f-b150-4d4e-b9fe-0c24e22fdb56/4390-4766

**Comment by swiatekm** [comment]
The agent not being healthy is due to beats receivers being put into a degraded state when they can't reach Elasticsearch. I think a simple fix would be to disable status reporting in these for now.

**Comment by belimawr** [comment]
I've had the same failure @khushijain21 reported in many of my PRs. Here is the last link I have:
- https://buildkite.com/elastic/elastic-agent/builds/30543/steps/canvas?jid=019a9ca9-5429-45d3-9d12-c7e90d6b4a4f#019a9ca9-5429-45d3-9d12-c7e90d6b4a4f/4387-4755

**Comment by swiatekm** [comment]
Saw a failure similar to the original report in https://buildkite.com/elastic/elastic-agent/builds/30665#019aa1af-113e-4fc0-9207-de6d38cfed1c. @pchila could this be a side effect of your recent fix?

**Comment by pchila** [comment]
> Saw a failure similar to the original report in https://buildkite.com/elastic/elastic-agent/builds/30665#019aa1af-113e-4fc0-9207-de6d38cfed1c. [@pchila](https://github.com/pchila) could this be a side effect of your recent fix?

@swiatekm my fix is impacting tests that rely on upgrade rollback when the agent is restarted a sufficient number of times during the grace period. The failure in your link points to such a test `TestStandaloneUpgradeRollbackOnRestarts` (although the issue points to a different test entirely).
The fix for the test you linked is in https://github.com/elastic/elastic-agent/pull/11324 which is merged and backported to 9.2
